### PR TITLE
Add project-scoped items for model mesh

### PR DESCRIPTION
**Pull request by ppadti**
Closes: [RHOAIENG-24572](https://issues.redhat.com/browse/RHOAIENG-24572)

## Description
This PR aims to support project-scoped serving runtime, hardware profile and accelerator profile for model mesh. Also create a re-usable component for acceleratorprofile-section.

1. Add model server with project-scoped Accelerator profile.

https://github.com/user-attachments/assets/5ae3af98-63ac-4af7-acc4-992df566ff77

Screenshot 2025-05-16 at 2 30 15 PM

Screenshot 2025-05-13 at 1 32 53 PM

2. with hardware profile

https://github.com/user-attachments/assets/915def47-3ba3-47f9-a635-d09f9b644de8

Screenshot 2025-05-12 at 2 19 22 PM

Screenshot 2025-05-12 at 2 17 56 PM

## How Has This Been Tested?
Try to add model server with project-scoped items(serving runtime, hardware profile and accelerator profile)

## Test Impact
Added cypress tests for the same. will be adding more for few edge cases in follow-up PR.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by pnaik1 — CHANGES_REQUESTED** [review]
great work! just a small nit

File: frontend/src/concepts/hardwareProfiles/useHardwareProfileConfig.ts [hunk 1]
```diff
@@ -105,6 +105,7 @@ export const useHardwareProfileConfig = (
   nodeSelector?: NodeSelector,
   visibleIn?: HardwareProfileFeatureVisibility[],
   namespace?: string,
+  hardwareProfileNamespace?: string | null,
```
Comment (pnaik1) [inline frontend/src/concepts/hardwareProfiles/useHardwareProfileConfig.ts:108]: I see now we are using annotation, doesnt the `hardwareProfileNamespace` and namespace give same value? can't we now replace namespace with `hardwareProfileNamespace`?
https://github.com/opendatahub-io/odh-dashboard/pull/4203/files#diff-0e61ca9f1a1c0dd011d4c2e693c5ced1687b2f220aeed574ad3787a7cb388e79R98-R130

**Review by pnaik1 — CHANGES_REQUESTED** [review]

File: frontend/src/concepts/hardwareProfiles/useHardwareProfileConfig.ts (hunk 1 above)
Comment (ppadti) [reply frontend/src/concepts/hardwareProfiles/useHardwareProfileConfig.ts:108]: yes, but when we update the deployment from project-scoped hardware profile to global-scoped hardware profile, then the profile will be selected from project-scoped list and not from global.

**Review by manaswinidas — APPROVED** [review]
/lgtm

Works fine. Will be good to go once the tests pass.

**Comment by pnaik1** [comment]
/lgtm
tested locally

File: frontend/src/pages/notebookController/screens/server/AcceleratorProfileSelectField.tsx [hunk 2]
```diff
@@ -312,16 +312,23 @@ const AcceleratorProfileSelectField: React.FC<AcceleratorProfileSelectFieldProps
...
-    return null;
-  }
-
   const filteredAcceleratorProfiles = getAcceleratorProfiles();
   const filteredDashboardAcceleratorProfiles = getDashboardAcceleratorProfiles();
   const hasProjectScopedAccelerators =
     isProjectScopedAvailable && currentProjectAcceleratorProfiles.length > 0;
 
+  // if there is more than a none option, show the dropdown
+  if (!isProjectScopedAvailable && options.length === 1) {
```
Comment (YuliaKrimerman) [inline frontend/src/pages/notebookController/screens/server/AcceleratorProfileSelectField.tsx:321]: Do we want to hide the component or maybe display no results found?

Comment (manaswinidas) [reply frontend/src/pages/notebookController/screens/server/AcceleratorProfileSelectField.tsx:321]: This is for a whole Accelerator dropdown, so I don't think "No results found" is necessary. When there are no accelerators the whole dropdown is hidden(this was the behavior before the "project-scoped" feature was introduced - we are following the same). Even when there are accelerators, it's not a required field, the user can choose to have "None" as accelerator even when there are accelerators. We should maybe add a skeleton in a follow-up issue.

**Review by manaswinidas — APPROVED** [review]
/lgtm

Tested all scenarios. Thanks for incorporating all the changes @ppadti
